src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,38 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    Animal.alive = Animal.alive.filter((animal) => animal.health > 0);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
```

Comment:
The `hide` method is implemented correctly, it sets the `hidden` property to `true` which is exactly what is required by the task. However, the method name could be misleading because it implies that the `hidden` property could be toggled (set to `false` if it's `true` and vice versa), but it only sets the `hidden` property to `true`. Consider renaming the method to something like `setHidden` or `makeHidden` to make it more clear that it only sets the `hidden` property to `true`.